runtime-sdk/src/crypto/signature/secp384r1.rs

```diff
@@ -0,0 +1,144 @@
+//! Secp384r1 signatures.
+use digest::{consts::U48, core_api::BlockSizeUser, Digest, FixedOutput, FixedOutputReset};
+use p384::{
+    self,
+    ecdsa::{
+        self,
+        signature::{DigestSigner as _, DigestVerifier, Signer as _, Verifier as _},
+    },
+};
+
+use crate::crypto::signature::{Error, Signature};
+
+/// A Secp384r1 public key (in compressed form).
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PublicKey(p384::EncodedPoint);
+
+impl PublicKey {
+    /// Return a byte representation of this public key.
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    /// Construct a public key from a slice of bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        let ep = p384::EncodedPoint::from_bytes(bytes).map_err(|_| Error::MalformedPublicKey)?;
+        if !ep.is_compressed() {
+            // This should never happen due to the size check above.
```

Comment:
Which size check above?
Can we remove this size check also on similar points?

Reply:
The size check I deleted but the hunk didn't get staged 😛 
The other curves implement a quick size check in `from_bytes`, so I updated them